code/obj/item/cigarette.dm

```diff
@@ -2,6 +2,136 @@
 /* ==================================================== */
 /* -------------------- Cigarettes -------------------- */
 /* ==================================================== */
+/obj/item/clothing/mask/cig_bundle
+	name = "bundle of cigarettes"
+	icon = 'icons/obj/items/cigarettes.dmi'
+	wear_image_icon = 'icons/mob/mask.dmi'
+	icon_state = "cig_bundle-2"
+	uses_multiple_icon_states = 1
+	item_state = "cig_bundle"
+	var/list/cig_list = list()
+	var/const/max_cigs = 21
+	var/on = 0
+	cant_self_remove = 1
+
+	New(var/obj/item/clothing/mask/cigarette/cig1, var/obj/item/clothing/mask/cigarette/cig2)
+		..()
+		cig1.set_loc(src)
+		cig2.set_loc(src)
+		if (cig1.on || cig2.on)
+			src.light()
+			src.visible_message("<span style='color:red'><b>[src] is set alight!</span>")
+
+	get_desc()
+		. += "There's [islist(contents) ? contents.len : "some unknown amount of"] cigarettes in there."
+
+	//This is basically just adding more cigs to the list
+	attackby(obj/item/W as obj, mob/user as mob)
+		if (istype(W, /obj/item/clothing/mask/cigarette))
+			var/obj/item/clothing/mask/cigarette/C = W
+			if (contents.len < max_cigs)
+				user.u_equip(C)
+				C.set_loc(src)
+				boutput(user, "You add [C] to [src].")
+				if (C.on && !src.on)		//light bundle if C is lit
+					src.light(user)
+				else if (!C.on && src.on)	//light C if bundle is lit
+					C.light(user)
+
+				update_icon()
+			else
+				boutput(user, "There's no way you can fit another cigarette in!")
+
+		else
+			//This is just a tricksy way to light the bundle without making this object a subtype of cigarette or copy-paste
+			//Pick a random object from the bundle's contents (should only be cigs sans shenanigans) and call its attackby proc with
+			//the arguments for this current proc. If it lights the cig, call the light proc on this object which lights all cigs.
+			if (islist(contents))
+				var/obj/item/clothing/mask/cigarette/cig = pick(contents)
+				if (istype(cig))
+					cig.attackby(W, user)
+					if (cig.on)
+						src.light(user)
+			else
+				..()
+
+	proc/update_icon()
+		if (islist(contents) && contents.len == 0)
+			return
+
+		//Simple math way you get numbers, almost done, but doesn't look great round((contents.len/max_cigs)*9)
+		var/state_num = 2
+		switch (contents.len)
+			if (2) 			state_num = 2
+			if (3) 			state_num = 3
+			if (4) 			state_num = 4
+			if (5) 			state_num = 5
+			if (6 to 7) 	state_num = 5
+			if (8 to 11) 	state_num = 6
+			if (12 to 15) 	state_num = 7
+			if (16 to 20) 	state_num = 8
+			if (21) 		state_num = 9
```

Comment:
Could have this be the default case, i.e. anything 21 or over gets this. Basically, have the cap be applied by the `max_cigs` var rather than the code.

Reply:
Yeah, I'll change that when I change this bit to reflect the sprites it uses.